API/Controllers/PaymentController.cs

```diff
@@ -17,30 +18,43 @@ public class PaymentController : ControllerBase
         // /payment/status/{order_id}
 
         private readonly IClusterClient _client;
+
         public PaymentController(IClusterClient client)
         {
             _client = client;
         }
 
         [HttpPost("pay/{user_id}/{order_id}")]
-        public Task<bool> Pay(Guid user_id, Guid order_id)
+        public async Task<bool> Pay(Guid user_id, Guid order_id)
         {
-            //POST - subtracts the amount of the order from the user’s credit(returns failure if credit is not enough)
-            //Payment grain  needed?
-            return Task.FromResult(true);
+            var user = _client.GetGrain<IUserGrain>(user_id);
+            var order = _client.GetGrain<IOrderGrain>(order_id);
+            var total = await order.GetTotalCost();
+
+            if (await user.ChangeCredit(-total)) return await order.Complete();
```

Comment:
If an order cannot be completed we need to rollback on the payment. 
Or we do it by hand, or if we use orleans transactions, using their annotations on the grain methods we're using should suffice. (We need to get the basics up and running to check the transactions issues)